luigi/contrib/datadog_metric.py

```diff
@@ -0,0 +1,126 @@
+        self._send_event(title=title, text=text, tags=event_tags, alert_type='error', priority='normal')
+
+    def handle_task_disabled(self, task, config):
+        title = "Luigi: A task has been disabled!"
+        lines = ['A task has been disabled in the pipeline named: {name}.']
+        lines.append('The task has failed {failures} times in the last {window}')
+        lines.append('seconds, so it is being disabled for {persist} seconds.')
+
+        preformated_text = ' '.join(lines)
+
+        text = preformated_text.format(name=task.family,
+                                       persist=config.disable_persist,
+                                       failures=config.retry_count,
+                                       window=config.disable_window)
+
+        tags = ["task_name:{name}".format(name=task.family)] + self._format_task_params_to_tags(task)
+
+        self._send_increment('task.disabled', tags=tags)
+
+        event_tags = tags + ["task_state:DISABLED"]
+        self._send_event(title=title, text=text, tags=event_tags, alert_type='error', priority='normal')
+
+    def handle_task_done(self, task):
+        # The task is already done -- Let's not re-create an event
+        if task.time_running is None:
+            return
+
+        title = "Luigi: A task has been completed!"
+        text = "A task has completed in the pipeline named: {name}".format(name=task.family)
+        tags = ["task_name:{name}".format(name=task.family)] + self._format_task_params_to_tags(task)
+
+        time_elapse = task.updated - task.time_running
+
+        self._send_increment('task.done', tags=tags)
+        self._send_gauge('task.execution_time', time_elapse, tags=tags)
+
+        event_tags = tags + ["task_state:DONE"]
+        self._send_event(title=title, text=text, tags=event_tags, alert_type='info', priority='low')
+
+    def _send_event(self, **params):
+        params['tags'] += self.default_tags()
+
+        api.Event.create(**params)
+
+    def _send_gauge(self, metric_name, value, tags=[]):
+        all_tags = tags + self.default_tags()
+
+        namespaced_metric = "{namespace}.{metric_name}".format(namespace=self._config.metric_namespace,
+                                                               metric_name=metric_name)
+        statsd.gauge(namespaced_metric, value, tags=all_tags)
+
+    def _send_increment(self, metric_name, value=1, tags=[]):
+        all_tags = tags + self.default_tags()
+
+        namespaced_metric = "{namespace}.{metric_name}".format(namespace=self._config.metric_namespace,
+                                                               metric_name=metric_name)
+        statsd.increment(namespaced_metric, value, tags=all_tags)
+
+    def _format_task_params_to_tags(self, task):
+        params = []
+        for key, value in task.params.items():
+            params.append("{key}:{value}".format(key=key, value=value))
+
+        return params
+
+    def default_tags(self):
```

Comment:
One of your last commits rose this to my attention: should this be a `@property`? It's utilizing `self._config.default_tags` which is a property. Sorry for not noticing sooner!

Reply:
Totally fair! Let me change that! :)